From 73ddc2474bcc5234fa0a4e406e015ed16d16b6c8 Mon Sep 17 00:00:00 2001
From: "sifram.rajas@gmail.com" <sifram.rajas@gmail.com>
Date: Fri, 2 Sep 2011 11:06:00 -0700
Subject: xhci: Redundant check in xhci_check_args for xhci->devs
Patch-mainline: v3.2
Git-commit: 73ddc2474bcc5234fa0a4e406e015ed16d16b6c8

The xhci_hcd->devs is an array of pointers rather than pointer to pointer.
Hence this check is not required.

Signed-off-by: Sifram Rajas <Sifram Rajas sifram.rajas@gmail.com>
Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/xhci.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -957,8 +957,7 @@ static int xhci_check_args(struct usb_hc
 
 	xhci = hcd_to_xhci(hcd);
 	if (check_virt_dev) {
-		if (!udev->slot_id || !xhci->devs
-			|| !xhci->devs[udev->slot_id]) {
+		if (!udev->slot_id || !xhci->devs[udev->slot_id]) {
 			printk(KERN_DEBUG "xHCI %s called with unaddressed "
 						"device\n", func);
 			return -EINVAL;
